tasks/collect-data/collect-data.yaml

```diff
@@ -201,3 +204,46 @@ spec:
             "$(workspaces.data.path)/$(params.subdirectory)/release_plan_admission.json"
 
         exit $RC
+    - name: print-pipeline-ref-info
+      image: quay.io/konflux-ci/release-service-utils:e633d51cd41d73e4b3310face21bb980af7a662f
+      script: |
+        #!/usr/bin/env bash
+        set +x
```

Comment:
`set +x` is the default, no? You already set your options line 212, I think 211 can be removed

Reply:
no. default with Tekton seems to be `set -x`

Reply:
No, it isn't. Here is output from your RHBA default tests (where we have `set -x` in the task)
```
[run-task : run-script] + set -ex
[run-task : run-script] + DEFAULT_ADVISORY_TYPE=RHBA
[run-task : run-script] + RESULTS_FILE=/workspace/data/results/create-advisory-results.json
```
That corresponds to
```
        set -ex

        DEFAULT_ADVISORY_TYPE="RHBA"
        RESULTS_FILE="$(workspaces.data.path)/$(params.resultsDirPath)/create-advisory-results.json"
```
in the task code. Notice the `set -x`.

Here is the output from the first step of collect-data from the tests of this very PR
```
[run-task : collect-data] {
[run-task : collect-data]     "apiVersion": "appstudio.redhat.com/v1alpha1",
[run-task : collect-data]     "kind": "Release",
[run-task : collect-data]     "metadata": {
```
This corresponds to
```
        #!/usr/bin/env bash
        set -eo pipefail
        RESULTS_DIR_PATH="results"
        if [ -n "$(params.subdirectory)" ]; then
          mkdir -p "$(workspaces.data.path)/$(params.subdirectory)"
          RESULTS_DIR_PATH="$(params.subdirectory)/results"
        fi
        mkdir "$(workspaces.data.path)/$RESULTS_DIR_PATH"
        echo -n "$RESULTS_DIR_PATH" > "$(results.resultsDir.path)"
        RELEASE_PATH="$(params.subdirectory)/release.json"
        echo -n "$RELEASE_PATH" > "$(results.release.path)"
        get-resource "release" "${RELEASE}" | tee "$(workspaces.data.path)/$RELEASE_PATH"
```
If `set -x` was the default, then you would see `RESULTS_DIR_PATH` being set and all the other things that you aren't seeing. You don't see any output until the `get-resource` line, which has `tee`

Reply:
with `set +x`

```
[run-task : print-pipeline-ref-info] + set +x
[run-task : print-pipeline-ref-info] Mock curl called with: -s https://api.github.com/repos/some-catalog/some-service-catalog/commits/development
[run-task : print-pipeline-ref-info] 
[run-task : print-pipeline-ref-info] Release Pipeline Ref Info:
[run-task : print-pipeline-ref-info] --------------------------
[run-task : print-pipeline-ref-info] {
[run-task : print-pipeline-ref-info]   "org": "some-catalog",
[run-task : print-pipeline-ref-info]   "repo": "some-service-catalog",
[run-task : print-pipeline-ref-info]   "revision": "development",
[run-task : print-pipeline-ref-info]   "pathinrepo": "pipelines/rh-advisories/rh-advisories.yaml",
[run-task : print-pipeline-ref-info]   "sha": "12345"
[run-task : print-pipeline-ref-info] }
```
without it

```
[run-task : print-pipeline-ref-info] + set -eu
[run-task : print-pipeline-ref-info] ++ jq -c .spec.pipeline.pipelineRef /workspace/data/db89058b-532a-4b35-93b7-f0a5c6d153bd/release_plan_admission.json
[run-task : print-pipeline-ref-info] + pipelineref='{"params":[{"name":"url","value":"https://github.com/some-catalog/some-service-catalog.git"},{"name":"revision","value":"development"},{"name":"pathInRepo","value":"pipelines/rh-advisories/rh-advisories.yaml"}],"resolver":"git"}'
[run-task : print-pipeline-ref-info] ++ jq -r '.resolver // ""'
[run-task : print-pipeline-ref-info] + resolver=git
[run-task : print-pipeline-ref-info] + '[' git == git ']'
[run-task : print-pipeline-ref-info] ++ jq -r '.params[] | select(.name=="url") | .value'
[run-task : print-pipeline-ref-info] + url=https://github.com/some-catalog/some-service-catalog.git
[run-task : print-pipeline-ref-info] ++ jq -r '.params[] | select(.name=="revision") | .value'
[run-task : print-pipeline-ref-info] + revision=development
[run-task : print-pipeline-ref-info] ++ jq -r '.params[] | select(.name=="pathInRepo") | .value'
[run-task : print-pipeline-ref-info] + pathinrepo=pipelines/rh-advisories/rh-advisories.yaml
[run-task : print-pipeline-ref-info] + orgrepo=some-catalog/some-service-catalog.git
[run-task : print-pipeline-ref-info] ++ echo some-catalog/some-service-catalog.git
[run-task : print-pipeline-ref-info] ++ cut -f1 -d/
[run-task : print-pipeline-ref-info] + org=some-catalog
[run-task : print-pipeline-ref-info] ++ echo some-catalog/some-service-catalog.git
[run-task : print-pipeline-ref-info] ++ cut -f2 -d/
[run-task : print-pipeline-ref-info] ++ cut -d. -f1
[run-task : print-pipeline-ref-info] + repo=some-service-catalog
[run-task : print-pipeline-ref-info] ++ curl -s https://api.github.com/repos/some-catalog/some-service-catalog/commits/development
[run-task : print-pipeline-ref-info] ++ echo 'Mock curl called with:' -s https://api.github.com/repos/some-catalog/some-service-catalog/commits/development
[run-task : print-pipeline-ref-info] Mock curl called with: -s https://api.github.com/repos/some-catalog/some-service-catalog/commits/development
[run-task : print-pipeline-ref-info] ++ echo -s https://api.github.com/repos/some-catalog/some-service-catalog/commits/development
[run-task : print-pipeline-ref-info] ++ jq -r '.sha // ""'
[run-task : print-pipeline-ref-info] ++ echo '{ "sha": "12345"}'
[run-task : print-pipeline-ref-info] + sha=12345
[run-task : print-pipeline-ref-info] + org=some-catalog
[run-task : print-pipeline-ref-info] + repo=some-service-catalog
[run-task : print-pipeline-ref-info] + revision=development
[run-task : print-pipeline-ref-info] + pathinrepo=pipelines/rh-advisories/rh-advisories.yaml
[run-task : print-pipeline-ref-info] + sha=12345
[run-task : print-pipeline-ref-info] + echo ''
[run-task : print-pipeline-ref-info] + echo 'Release Pipeline Ref Info:'
[run-task : print-pipeline-ref-info] + echo --------------------------
[run-task : print-pipeline-ref-info] 
[run-task : print-pipeline-ref-info] Release Pipeline Ref Info:
[run-task : print-pipeline-ref-info] --------------------------
[run-task : print-pipeline-ref-info] ++ jq -n -c --arg org some-catalog --arg repo some-service-catalog --arg revision development --arg pathinrepo pipelines/rh-advisories/rh-advisories.yaml --arg sha 12345 '$ARGS.named | tostring'
[run-task : print-pipeline-ref-info] + json='"{\"org\":\"some-catalog\",\"repo\":\"some-service-catalog\",\"revision\":\"development\",\"pathinrepo\":\"pipelines/rh-advisories/rh-advisories.yaml\",\"sha\":\"12345\"}"'
[run-task : print-pipeline-ref-info] + echo '"{\"org\":\"some-catalog\",\"repo\":\"some-service-catalog\",\"revision\":\"development\",\"pathinrepo\":\"pipelines/rh-advisories/rh-advisories.yaml\",\"sha\":\"12345\"}"'
[run-task : print-pipeline-ref-info] + jq .
[run-task : print-pipeline-ref-info] ++ echo '"{"org":"some-catalog","repo":"some-service-catalog","revision":"development","pathinrepo":"pipelines/rh-advisories/rh-advisories.yaml","sha":"12345"}"'
[run-task : print-pipeline-ref-info] ++ sed 's/^"//g'
[run-task : print-pipeline-ref-info] ++ sed 's/"$//g'
[run-task : print-pipeline-ref-info] {
[run-task : print-pipeline-ref-info]   "org": "some-catalog",
[run-task : print-pipeline-ref-info]   "repo": "some-service-catalog",
[run-task : print-pipeline-ref-info]   "revision": "development",
[run-task : print-pipeline-ref-info]   "pathinrepo": "pipelines/rh-advisories/rh-advisories.yaml",
[run-task : print-pipeline-ref-info]   "sha": "12345"
[run-task : print-pipeline-ref-info] }

```

Reply:
Just to be clear, the [documentation](https://tekton.dev/docs/pipelines/tasks/#running-scripts-within-steps) says this:

> Scripts that do not start with a [shebang](https://en.wikipedia.org/wiki/Shebang_(Unix)) line will have the following default preamble prepended:
> ```
> #!/bin/sh
> set -e
> ```

What happens here though is that the mock has `set -eux` and then this script has `set +x`. So when testing, the mock is injected on top of the step script and it will first enable tracing and then you disable it. So if you're worried you will have all these extra logs when using the task, don't be. But please use the same `-e` and `-u` setting in the task and in the mock (I don't care about `-x`), so that the test is realistic. Currently it differs.